File: контест 2/A.cpp
#include <iostream>
#include <vector>
#include <queue>

struct line{
    int left;
    int right;
    bool operator < (const line & other) const {
        return this->right > other.right;
    }
};


bool cmp(const line & arg1, const line & arg2){
    return arg1.left < arg2.left;
}

int
main(void){
    int n, m;
    std::cin >> n >> m;
    std::vector<int>ans(n);
    std::vector<line> lines;
    for (int i = 0; i < m; i++){
        int l, r;
        std::cin >> l >> r;
        line line = {l, r};
        lines.push_back(line);
    }

    std::sort(lines.begin(), lines.end(), cmp);
    std::priority_queue<line> priority;
    int cur = 0;
    for (int i = 1; i < n + 1; i ++){
        while (cur < m && lines[cur].left == i){
            priority.push(lines[cur]);
            cur ++;
        }
        while ((priority.size() > 0) && priority.top().right < i){
            priority.pop();
        }
        ans[i - 1] = priority.size();
    }

    for (int i = 0; i < ans.size(); i++){
        std::cout << ans[i] << " ";
    }
    std::cout << std::endl;

}

File: контест 2/A.py
import queue

n, m = map(int, input().split())
ans = [0] * n


line = []
for _ in range(m):
    l, r = map(int, input().split())
    line.append([l, r])

cur = 0
priority = queue.PriorityQueue(maxsize=m + 1)
line.sort()
for i in range(1, n + 1):
    while cur < m and line[cur][0] == i:
        priority.put(line[cur][1], line[cur])
        cur += 1
    while not(priority.empty()) and priority.queue[0] < i:
        item = priority.get()
    ans[i - 1] = priority.qsize()
    



print(*ans)